spring-integration-core/src/test/java/org/springframework/integration/config/InvalidChannelWithCapacityAndMessageStoreParserTests.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2002-2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.config;
+
+import org.hamcrest.Matchers;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import org.springframework.beans.factory.parsing.BeanDefinitionParsingException;
+import org.springframework.context.support.ClassPathXmlApplicationContext;
+
+/**
+ * @author Manuel Jordan
+ */
+public class InvalidChannelWithCapacityAndMessageStoreParserTests {
```

Comment:
Look. We are going to test that our attributes restriction works well showing an error on parsing the XML. Therefore it fully fails the entire application context.
That's why the code in the test-method looks like:
```
new ClassPathXmlApplicationContext(getClass().getSimpleName() + "-context.xml", getClass()).close();
```
But having the `@Rule ExpectedException` we definitely try to be sure that the `@Test` will pass with desired error.
Those new `@Test` to the same class won't break anything, each test-method is isolated from any other with its own attempt to load its own context.
For example `ChainParserTests#invalidNestedChainWithLoggingChannelAdapter()`.